hack/lib/version.sh

```diff
@@ -27,6 +27,26 @@
 #    KUBE_GIT_MAJOR - The major part of the version
 #    KUBE_GIT_MINOR - The minor component of the version
 
+# Ensure GNU tar is present.
+function kube::version::ensure_gnu_date() {
+  if [[ -n "${GNU_DATE:-}" ]]; then
+    return
+  fi
+  GNU_DATE="date"
+  if which gdate &>/dev/null; then
+      GNU_DATE="gdate"
+  elif which gnudate &>/dev/null; then
+      GNU_DATE="gnudate"
+  fi
+  if ! "${GNU_DATE}" --version | grep -q GNU; then
+    echo "  !!! Cannot find GNU tar. Build on Linux or install GNU tar"
+    echo "      on Mac OS X (brew install coreutils)."
+    return 1
```

Comment:
still not sure if this should be a hard failure or just a warning, it's only necessary if SOURCE_DATE_EPOCH so maybe only fail if SOURCE_DATE_EPOCH ?